cgd_tx_eff/src/edu/ohsu/compbio/txeff/tx_eff_vcf.py

```diff
@@ -0,0 +1,296 @@
+    TFX_VARIANT_EFFECT = 'TFX_VARIANT_EFFECT'
+    TFX_VARIANT_TYPE = 'TFX_VARIANT_TYPE'
+    TFX_BASE_POSITION = 'TFX_BASE_POSITION'
+    TFX_PROTEIN_TRANSCRIPT = 'TFX_PROTEIN_TRANSCRIPT'
+    TFX_EXON = 'TFX_EXON'
+    TFX_AMINO_ACID_POSITION = 'TFX_AMINO_ACID_POSITION'
+     
+def _parse_args():
+    '''
+    Validate and return command line arguments.
+    '''
+    parser = argparse.ArgumentParser(description='Produce a VCF containing transcript details generated by tx_eff.hgvs.py')
+
+    parser.add_argument('-i', '--in_vcf', 
+                    help='Input VCF', 
+                    type=argparse.FileType('r'), 
+                    required=True)
+
+    parser.add_argument('-o', '--out_vcf', 
+                    help='Output VCF', 
+                    type=argparse.FileType('w'), 
+                    required=True)
+
+    parser.add_argument('-c', '--in_csv', 
+                    help='Input CSV', 
+                    type=argparse.FileType('r'), 
+                    required=True)
+    
+    parser.add_argument('--version', action='version', version='%(prog)s ' + VERSION)
+    
+    args = parser.parse_args()
+    
+    return args
+    
+    args = parser.parse_args()
+
+
+def _read_vcf(vcf_filename: str):
+    '''
+    Read each variant from the VCF file and return a map of Variant[vcf-record]. Each variant must be unique and have only one ALT allele.  
+    '''
+    vcf_reader = vcfpy.Reader.from_path(vcf_filename)
+    variant_dict = dict()
+    
+    for vcf_record in vcf_reader:
+        if len(vcf_record.ALT) > 1:
+            raise Exception(f'VCF variants must have just one ALT allel: {vcf_record.CHROM}-{vcf_record.POS}-{vcf_record.REF}-{vcf_record.ALT}')
+        
+        variant = Variant(vcf_record.CHROM, vcf_record.POS, vcf_record.REF, vcf_record.ALT[0].value)
+
+        if variant_dict.get(variant):
+            raise Exception(f"Duplicate variant in {vcf_filename}: {variant}")
+        else:
+            variant_dict[variant] = vcf_record
+        
+    vcf_reader.close()
+    
+    return variant_dict, vcf_reader.header
+
+
+def _noneIfEmpty(value: str):
+    '''
+    Return None if the string is an empty string.
+    ''' 
+    if value == '':
+        return None
+    return value
+
+def _get_transcripts_dict(transcripts: list):
+    '''
+    Return a dictionary where the key is a variant and the value is a list of transcripts 
+    ''' 
+    transcript_dict = defaultdict(list)
+    
+    for transcript in transcripts:        
+        variant = Variant(transcript.chromosome, transcript.position, transcript.reference, transcript.alt)
+        transcript_dict[variant].append(transcript)
+        logger.debug(f"{variant} has {len(transcript_dict[variant])} transcripts")
+
+    return transcript_dict
+
+def _add_transcripts_to_vcf(vcf_variant_dict, transcript_dict):
+    '''
+    Transform the transcript details into parallel arrays and add them to each VCF row.  
+    '''
+    vcf_records = list()
```

Comment:
I think, in general, you would say vcf_records = [] is more pythonic than list().  Also, this guy says it's faster:

https://towardsdatascience.com/no-and-list-are-different-in-python-8940530168b0#:~:text=As%20shown%20in%20the%20results,as%20%7B%7D%20and%20dict()%20.